test/loki_service.cc

```diff
@@ -1,6 +1,10 @@
 #include "test.h"
 #include <cstdint>
 
+// the utility wrapper around rapidjson parses into a boost::ptree which preserves
+// order. We are comparing json which is unordered so lets skip the rapidjson wrapper
+#include <rapidjson/document.h>
```

Comment:
you shouldnt include rapidjson directly as our wrapper below uses some macro tricks to replace its `assert`s with `throw`s. i supose in a unit test it may not matter, but i would prefer rapidjson to throw an exception in my unit test if i expected it to. the wrapper below does include this header anyway so you can still use rapidjson as normal and avoid any strang asserts in the place of a more usable `throw`. again might not matter for this unit test but its something to remember when doing rapidjson in the rest of the library.

Reply:
Include removed.